File: day18/ConsecutiveSequence.java
import java.util.*;

public class Day18 {
    public static int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums) numSet.add(num);

        int longest = 0;
        for (int num : numSet) {
            if (!numSet.contains(num - 1)) {
                int current = num;
                int streak = 1;

                while (numSet.contains(current + 1)) {
                    current++;
                    streak++;
                }

                longest = Math.max(longest, streak);
            }
        }
        return longest;
    }

    public static void main(String[] args) {
        Random rand = new Random();
        int length = rand.nextInt(21) + 10; // length 10 to 30
        Set<Integer> numSet = new HashSet<>();
        while (numSet.size() < length) {
            numSet.add(rand.nextInt(100));
        }
        int[] nums = numSet.stream().mapToInt(Integer::intValue).toArray();

        System.out.println("Input: " + Arrays.toString(nums));
        System.out.println("Longest consecutive sequence length: " + longestConsecutive(nums));
    }
}

File: day18/consecutive_sequence.rs
use rand::{thread_rng, Rng};
use std::collections::HashSet;

fn longest_consecutive(nums: &[i32]) -> usize {
    let num_set: HashSet<i32> = nums.iter().cloned().collect();
    let mut longest = 0;

    for &num in &num_set {
        if !num_set.contains(&(num - 1)) {
            let mut current = num;
            let mut streak = 1;

            while num_set.contains(&(current + 1)) {
                current += 1;
                streak += 1;
            }

            longest = longest.max(streak);
        }
    }

    longest
}

fn main() {
    let mut rng = thread_rng();
    let length = rng.gen_range(10..=30);

    let mut unique = HashSet::new();
    while unique.len() < length {
        unique.insert(rng.gen_range(0..100));
    }

    let nums: Vec<i32> = unique.into_iter().collect();
    println!("Input: {:?}", nums);
    println!(
        "Longest consecutive sequence length: {}",
        longest_consecutive(&nums)
    );
}
